forte/processors/data_augment/algorithms/embedding_similarity_replacement_op.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import random
+
+from typing import Tuple
+import numpy as np
+from texar.torch.data import Embedding
+from texar.torch.data import Vocab
+
+from ft.onto.base_ontology import Annotation
+from forte.common.configuration import Config
+from forte.processors.data_augment.algorithms.text_replacement_op \
+    import TextReplacementOp
+from forte.processors.data_augment.utils.utils import l2_norm
+
+__all__ = [
+    "EmbeddingSimilarityReplacementOp",
+]
+
+random.seed(0)
+
+emb_types = ['glove']
+
+
+class EmbeddingSimilarityReplacementOp(TextReplacementOp):
+    r"""
+    This class is a replacement op leveraging pre-trained word
+    embeddings, such as word2vec and glove, to replace the input
+    word with another word with similar word embedding.
+    By default, the replacement word is randomly chosen from the
+    top k words with the most similar embeddings.
+
+    Args:
+        embedding: A texar.tf.data.Embedding object. Can be initialized
+            from pre-trained embedding file using helper functions
+            E.g. forte.processors.data_augment.utils.utils.load_glove_embedding
+        vocab: A texar.torch.data.Vocab object. Can be initialized from
+            pre-trained embedding file using helper functions
+            E.g. forte.processor.data_augment.utils.utils.load_glove_vocab
+        configs: The config should contain `top_k`,
```

Comment:
Let's use the standard forte way of adding `default_configs`. Don't add it here.

Reply:
I'm a bit confused. The `replacement_ops` cannot be directly initialized by `initialize` since it is a parameter to the processors. I looked at other `replacement_ops` and they all use a `configs` parameter instead of `default_configs`.

Reply:
I see. Then we simply follow the other's convention now. I may have forgotten some details about this design.

One thing about the config is whether we provide a default_config (which restricts what config parameters to be provided). Here we may also add a default config (manually)?

Reply:
We pass parameters to `replacement_ops` using the `configs`. In an actual pipeline, this op would be initialized by passing the `data_aug_op_config` to the processor's config (see the pipeline test)